CMakeLists.txt

```
@@ -17,6 +17,7 @@ ament_vendor(spdlog_vendor
   CMAKE_ARGS
     -DSPDLOG_BUILD_TESTS:BOOL=OFF
     -DSPDLOG_BUILD_EXAMPLE:BOOL=OFF
+    "-DCMAKE_CXX_FLAGS=-Wno-deprecated-declarations"
```

We shouldn't ignore any already specified `CMAKE_CXX_FLAGS`.

```suggestion
    "-DCMAKE_CXX_FLAGS=-Wno-deprecated-declarations ${CMAKE_CXX_FLAGS}"
```

---

That's a good point, but I also think we should invert the order to:

```
"-DCMAKE_CXX_FLAGS=-Wno-deprecated-declarations ${CMAKE_CXX_FLAGS}"
```

The reason being if the CMAKE_CXX_FLAGS contains `-Wdeprecated-declarations`, it will re-enable it (we recently ran into this elsewhere).

---

> if the CMAKE_CXX_FLAGS contains `-Wdeprecated-declarations`, it will re-enable it 

The `CMAKE_CXX_FLAGS` in the vendor package is defined by us, not the payload. By putting it after CMAKE_CXX_FLAGS we're doing exactly what other projects do that forces us to patch - giving the developer no opportunity to override our flags.

The developer should have final say. If anything is in `CMAKE_CXX_FLAGS` here, somebody put it there on purpose.

---

> The `CMAKE_CXX_FLAGS` in the vendor package is defined by us, not the payload.

Ah, that's a good point.  OK, in that case I agree with your original suggestion then.